src/routes/sessions/[id]/+page.svelte

```diff
@@ -44,20 +43,18 @@
 	let knowledge: Knowledge | null;
 	let session: Session;
 	let completion: string;
+	let abortController: AbortController;
 	let prompt: string;
-	let promptCached: string;
 	let promptTextarea: HTMLTextAreaElement;
+	let isCompletionInProgress = false;
 	let isPromptFullscreen = false;
 	let shouldFocusTextarea = false;
 	let userScrolledUp = false;
-	let ollamaInstance: Ollama | null = null;
-	let isWaitingForResponse = false;
 
 	const shouldConfirmDeletion = writable(false);
 
 	$: session = loadSession(data.id);
 	$: isNewSession = !session?.messages.length;
-	$: isLastMessageFromUser = session?.messages[session.messages.length - 1]?.role === 'user';
```

Comment:
`isLastMessageFromUser` was only used to determine if there was a completion in progress so we are now using `isCompletionInProgress`